aregion.cpp

```diff
@@ -29,14 +29,12 @@ Location *GetUnit(std::list<Location *>& locs, int unitid)
     return nullptr;
 }
 
-Farsight::Farsight()
+Farsight::Farsight(Faction *faction, Unit *unit, int level, int observation)
+    : faction(faction), unit(unit), level(level), observation(observation)
 {
-    faction = 0;
-    unit = 0;
-    level = 0;
-    observation = 0;
-    for (int i = 0; i < NDIRS; i++)
+    for (int i = 0; i < NDIRS; i++) {
```

Comment:
Just stylistically, this is one place where I would have just put it onto one line and avoided the parens entirely.

Reply:
My own preference goes towards "making it hard for future editors to make a mistake". Even if you write it in a single line, there's no guarantee that someone's code formatter breaks it into two in the future, and next thing you know, you get the classic Apple `goto fail;` problem. [Reference](https://www.blackduck.com/blog/understanding-apple-goto-fail-vulnerability-2.html)